frontend/app/src/composables/history/events/mapping.ts

```diff
@@ -0,0 +1,249 @@
+      const eventSubtypeNormalized = eventSubtype || 'none';
+
+      if (location && counterparty) {
+        const subTypesFromPerProtocolMapping = get(
+          historyEventTypePerProtocolMapping
+        )[location]?.[counterparty]?.[eventTypeNormalized]?.[
+          eventSubtypeNormalized
+        ];
+
+        if (subTypesFromPerProtocolMapping) {
+          return subTypesFromPerProtocolMapping;
+        }
+      }
+
+      return (
+        get(historyEventTypeGlobalMapping)[eventTypeNormalized]?.[
+          eventSubtypeNormalized
+        ] ?? undefined
+      );
+    });
+
+  const getEventTypeData = (
+    event: {
+      eventType?: string | null;
+      eventSubtype?: string | null;
+      counterparty?: string | null;
+      location?: string | null;
+    },
+    showFallbackLabel = true
+  ): ComputedRef<ActionDataEntry> =>
+    computed(() => {
+      const type = get(getEventType(event));
+
+      if (type) {
+        return get(transactionEventTypesData).find(
+          (data: ActionDataEntry) =>
+            data.identifier.toLowerCase() === type.toLowerCase()
+        )!;
+      }
+
+      const unknownLabel = tc('backend_mappings.events.type.unknown');
+      const label = showFallbackLabel
+        ? event.eventSubtype || event.eventType || unknownLabel
+        : unknownLabel;
+
+      return {
+        identifier: '',
+        label,
+        icon: 'mdi-help',
+        color: 'red'
+      };
+    });
+
+  const getEventCounterpartyData = (
+    event: HistoryEventEntry,
+    scrambler?: (hex: string) => string
+  ): ComputedRef<ActionDataEntry | null> =>
+    computed(() => {
+      const { counterparty } = event;
+      const excludedCounterparty = ['gas'];
+
+      if (counterparty && excludedCounterparty.includes(counterparty)) {
+        return null;
+      }
+
+      if (counterparty && !isValidEthAddress(counterparty)) {
+        const data = get(historyEventCounterpartiesData).find(
+          (data: ActionDataEntry) => {
+            if (data.matcher) {
+              return data.matcher(counterparty);
+            }
+            return data.identifier.toLowerCase() === counterparty.toLowerCase();
+          }
+        );
+
+        if (data) {
+          return {
+            ...data,
+            label: counterparty.toUpperCase()
+          };
+        }
+
+        return {
+          identifier: '',
+          label: counterparty,
+          icon: 'mdi-help',
+          color: 'red'
+        };
+      }
+
+      if (!counterparty && !event.address) {
+        return null;
+      }
+
+      const counterpartyAddress =
+        scrambler?.(counterparty || event.address || '') ??
+        counterparty ??
+        event.address;
+
+      return {
+        identifier: '',
+        label: counterpartyAddress || ''
+      };
+    });
+
+  const counterparties: ComputedRef<string[]> = computed(() =>
+    get(historyEventCounterpartiesData).map(({ identifier }) => identifier)
+  );
```

Comment:
can't we use the [useArrayMap](https://vueuse.org/shared/useArrayMap/#usearraymap) here from vueuse?